src/Npgsql/NpgsqlConnectionStringBuilder.cs

```diff
@@ -34,6 +34,11 @@ public sealed class NpgsqlConnectionStringBuilder : DbConnectionStringBuilder, I
         /// </summary>
         static readonly Dictionary<PropertyInfo, object> PropertyDefaults;
 
+        /// <summary>
+        /// Cached DataSource value to reduce allocations on NpgsqlConnection.DataSource.get
+        /// </summary>
+        internal string DataSourceCached { get; set; }
```

Comment:
Why not just name this `DataSource`?

Reply:
As mentioned below, I wanted to express intention that this value is a cached value.